lib/client/config.go

```diff
@@ -0,0 +1,343 @@
+	} else {
+		return n.Enabled, nil
+	}
+}
+
+// SetDefaultASNumber sets the default AS Number used by the BGP agent running
+// on each node.  This may be overridden by an explicitly configured value in
+// the node resource.
+func (c *config) SetDefaultASNumber(asNumber uint64) error {
+	_, err := c.c.backend.Apply(&model.KVPair{
+		Key:   model.GlobalBGPConfigKey{Name: "as_num"},
+		Value: strconv.FormatUint(asNumber, 10),
+	})
+	return err
+}
+
+// SetDefaultASNumber gets the default AS Number used by the BGP agent running
+// on each node.  See SetDefaultASNumber for more details.
+func (c *config) GetDefaultASNumber() (uint64, error) {
+	if s, err := c.getValue(model.GlobalBGPConfigKey{Name: "as_num"}); err != nil {
+		return 0, err
+	} else if s == nil {
+		return 64511, nil
+	} else if asn, err := strconv.ParseUint(*s, 10, 64); err != nil {
+		return 64511, err
+	} else {
+		return asn, nil
+	}
+}
+
+// SetDefaultIPIP sets the default IP in IP enabled setting inherited by all nodes
+// in the Calico cluster.  When IP in IP is enabled, packets routed to IP addresses
+// that fall within an IP in IP enabled Calico IP Pool, will be routed over an
+// IP in IP tunnel.
+func (c *config) SetDefaultIPIP(enabled bool) error {
+	_, err := c.c.backend.Apply(&model.KVPair{
+		Key:   model.GlobalConfigKey{Name: "IpInIpEnabled"},
+		Value: strconv.FormatBool(enabled),
+	})
+	return err
+}
+
+// GetDefaultIPIP gets the default IPIP enabled setting.  See SetDefaultIPIP for details.
+func (c *config) GetDefaultIPIP() (bool, error) {
+	if s, err := c.getValue(model.GlobalConfigKey{Name: "IpInIpEnabled"}); err != nil {
+		return false, err
+	} else if s == nil {
+		return false, nil
+	} else if enabled, err := strconv.ParseBool(*s); err != nil {
+		return false, err
+	} else {
+		return enabled, nil
+	}
+}
+
+// SetNodeIPIPTunnelAddress sets the IP in IP tunnel address for a specific node.
+// Felix will use this to configure the tunnel.
+func (c *config) SetNodeIPIPTunnelAddress(node string, ip *net.IP) error {
+	key := model.HostConfigKey{Hostname: node, Name: "IpInIpTunnelAddr"}
+	if ip == nil {
+		err := c.deleteConfig(key)
+		return err
+	} else {
+		_, err := c.c.backend.Apply(&model.KVPair{
+			Key:   key,
+			Value: ip.String(),
+		})
+		return err
+	}
+}
+
+// GetNodeIPIPTunnelAddress gets the IP in IP tunnel address for a specific node.
+// See SetNodeIPIPTunnelAddress for more details.
+func (c *config) GetNodeIPIPTunnelAddress(node string) (*net.IP, error) {
+	ip := &net.IP{}
+	if s, err := c.getValue(model.HostConfigKey{Hostname: node, Name: "IpInIpTunnelAddr"}); err != nil {
+		return nil, err
+	} else if s == nil {
+		return nil, nil
+	} else if err = ip.UnmarshalText([]byte(*s)); err != nil {
+		return nil, err
+	} else {
+		return ip, nil
+	}
+}
+
+// SetDefaultLogLevel sets the system default log level used by the node.  This
+// may be overridden on a per-node basis.
+func (c *config) SetDefaultLogLevel(level LogLevel) error {
```

Comment:
Think I'd prefer Global instead of Default (since Default, to me, means "what you get if you don't have any value set").